core/http/src/parse/uri/tables.rs

```diff
@@ -29,6 +29,40 @@ const PATH_CHARS: [u8; 256] = [
         0,     0,     0,     0,     0,     0                              // 25x
 ];
 
+
+const QUERY_CHARS: [u8; 256] = [
+    //  0      1      2      3      4      5      6      7      8      9
+        0,     0,     0,     0,     0,     0,     0,     0,     0,     0, //   x
+        0,     0,     0,     0,     0,     0,     0,     0,     0,     0, //  1x
+        0,     0,     0,     0,     0,     0,     0,     0,     0,     0, //  2x
+        0,     0,     0,  b'!',     0,     0,  b'$',  b'%',  b'&',     0, //  3x
+     b'(',  b')',  b'*',  b'+',  b',',  b'-',  b'.',  b'/',  b'0',  b'1', //  4x
+     b'2',  b'3',  b'4',  b'5',  b'6',  b'7',  b'8',  b'9',  b':',  b';', //  5x
+//      <             > (we allow this chars: while they are not valid in URLs but in URI)
```

Comment:
The comment is confusing and maybe wrong - I don't know what "not valid in URLs but in URI" means.

But note that query might need the same distinction that path has between "valid always" vs "valid only in a route specification as a placeholder," which is why `<` and `>` are handled specially there.

Reply:
@jebrosen I mean that it is correct for declaration in usage code, but not in HTTP request